Classes/Netlogix/Cqrs/Command/CommandBus.php

```diff
@@ -40,14 +40,27 @@ class CommandBus {
 
 	/**
 	 * @param CommandInterface $command
+	 * @throws \Exception
 	 */
 	public function delegate(CommandInterface $command) {
 		$this->initializeCommandHandlers();
-		foreach ($this->commandHandlers as $commandHandler) {
-			if ($commandHandler->canHandle($command)) {
-				$commandHandler->handle($command);
+		try {
+			foreach ($this->commandHandlers as $commandHandler) {
+				if ($commandHandler->canHandle($command)) {
+					$commandHandler->handle($command);
+				}
 			}
+		} catch (\Exception $e) {
+			$this->logCommand($command);
```

Comment:
Thinking about it some more, it would make sense to log the exception (exception class, message, file, line, maybe just serialize the exception?), too.